packages/cmrjs/tests/cmr-utils/test-mergeURLs.js

```diff
@@ -0,0 +1,215 @@
+const test = require('ava');
+const rewire = require('rewire');
+
+const cmrUtil = rewire('../../cmr-utils');
+const mergeURLs = cmrUtil.__get__('mergeURLs');
+
+const sortByURL = (a, b) => a.URL < b.URL;
+
+
+test('Merges two sets of URLs.', (t) => {
+  const originalURLs = [
+    {
+      URL: 'https://path/to/metadata.cmr.xml',
+      URLDescription: 'File to download'
+    }
+  ];
+  const newURLs = [
+    {
+      URL: 's3://path/to/different-file.hdf',
+      URLDescription: 'File to download'
+    }
+  ];
+  const expected = [...originalURLs, ...newURLs];
+
+  const actual = mergeURLs(originalURLs, newURLs);
+
+  t.deepEqual(expected.sort(sortByURL), actual.sort(sortByURL));
+});
+
+test('Replaces an updated URL.', (t) => {
+  const originalURLs = [
+    {
+      URL: 'https://path/to/metadata.cmr.xml',
+      URLDescription: 'File to download'
+    }
+  ];
+  const newURLs = [
+    {
+      URL: 's3://path/to/metadata.cmr.xml',
+      URLDescription: 'File to download'
+    }
+  ];
+  const expected = [...newURLs];
+
+  const actual = mergeURLs(originalURLs, newURLs);
+
+  t.deepEqual(expected.sort(sortByURL), actual.sort(sortByURL));
+});
+
+test('Removes a file made private.', (t) => {
+  const originalURLs = [
+    {
+      URL: 'https://path/to/metadata.cmr.xml',
+      URLDescription: 'File to download'
+    }
+  ];
+  const newURLs = [];
+  const deletedUrls = [
+    { URL: 'https://now/private/metadata.cmr.xml' }
+  ];
+
+  const expected = [];
+
+  const actual = mergeURLs(originalURLs, newURLs, deletedUrls);
+
+  t.deepEqual(expected.sort(sortByURL), actual.sort(sortByURL));
+});
+
+test('Replaces an updated URL, but keeps any additional metadata from the original urlObject.', (t) => {
```

Comment:
Is this something we can expect to never actually see happen in production, with the API move? Or is there some scenario where that move generates URL metadata?

Reply:
Same question regarding the below `'Replaces an updated URL, but does not overwrite existing metadata fields.'` test.

Reply:
So when we move a file the only url metadata fields we touch are the URL, Type, and Description fields.  But if the metadata already had something like MimeType, or Size or something, we don't want to change that just because we moved the file.  This tests shows we preserve any of that metadata on a merge.

Reply:
Maybe I answered the wrong question, we do update the URL metadata in the new object every time we move, but don't want to disturb any existing metadata that is already in this urlObject.

Am I close?

Reply:
Is the intent for it never to overwrite any existing metadata other than the URL itself?

Reply:
`// preferring all metadata from original except the new url.URL` I see now, OK 👍

Reply:
Yeah, that could be wrong, but that's what lauren and I came up with, it's explicit if someone doesn't like it we can fix it. Thanks.

Reply:
Yeah we only want to update URLs, and only the URLs that we change. There could be additional links in the metadata file, pointing to the something like the DAAC website or something. We wouldn't want to mess with that at all.